src/test/perf_counters.cc

```diff
@@ -219,3 +222,54 @@ TEST(PerfCounters, ResetPerfCounters) {
   ASSERT_EQ("", client.do_request("{ \"prefix\": \"perf reset\", \"var\": \"test_perfcounter_1\", \"format\": \"json\" }", &msg));
   ASSERT_EQ(sd("{\"error\":\"Not find: test_perfcounter_1\"}"), msg);
 }
+
+enum {
+  TEST_PERFCOUNTERS3_ELEMENT_FIRST = 800,
+  TEST_PERFCOUNTERS3_ELEMENT_1,
+  TEST_PERFCOUNTERS3_ELEMENT_2,
+  TEST_PERFCOUNTERS3_ELEMENT_LAST,
+};
+
+static PerfCounters* setup_test_perfcounters3(CephContext *cct)
+{
+  PerfCountersBuilder bld(cct, "test_perfcounter_3",
+          TEST_PERFCOUNTERS3_ELEMENT_FIRST, TEST_PERFCOUNTERS3_ELEMENT_LAST);
+  bld.add_time_avg(TEST_PERFCOUNTERS3_ELEMENT_1, "mono_duration");
+  bld.add_time_avg(TEST_PERFCOUNTERS3_ELEMENT_2, "utime_t");
+  return bld.create_perf_counters();
+}
+
+TEST(SteadyClock, PerfCounter) {
+  PerfCountersCollection *coll = g_ceph_context->get_perfcounters_collection();
+  PerfCounters* fake_pf = setup_test_perfcounters3(g_ceph_context);
+  coll->add(fake_pf);
+
+  AdminSocketClient client(get_rand_socket_path());
+  std::string msg;
+  ASSERT_EQ("", client.do_request("{ \"prefix\": \"perf dump\", \"format\": \"json\" }", &msg));
+
+  mono_time t1, t2;
+  mono_clock::duration elapsed;
+  utime_t u_t1, u_t2, u_elapsed;
+
+  t1 = mono_clock::now();
+  u_t1 = ceph_clock_now();
+
+  sleep(1);
```

Comment:
I'll nudge you toward std::this_thread::sleep_for() and std::this_thread::sleep_until(), which unlike sleep() are standard.